pkg/android/get-android-info.go

```diff
@@ -0,0 +1,245 @@
+				defer destinationFile.Close()
+
+				_, err = io.Copy(destinationFile, zippedFile)
+				if err != nil {
+					return "", err
+				}
+
+				return outputPath, nil
+			}
+		}
+	}
+	return "", fmt.Errorf("no aab file")
+}
+
+// GetAndroidXMLData - Pulls information from a human-readable xml file into a struct
+func GetAndroidXMLData(manifestFile string) (*AndroidManifestData, error) {
+	data, err := os.ReadFile(manifestFile)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read " + manifestFile + " : " + err.Error())
+	}
+
+	var manifestData *AndroidManifestData
+	err = xml.Unmarshal(data, &manifestData)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse data from " + manifestFile + " : " + err.Error())
+	}
+
+	return manifestData, nil
+}
+
+func GetAndroidProtobufData(path string) (*AndroidManifestData, error) {
+	aabManifestData := make(map[string]string)
+
+	content, err := os.ReadFile(path)
+
+	if err != nil {
+		return nil, err
+	}
+
+	rawAabManifestData := &proto_messages.XmlNode{}
+
+	err = proto.Unmarshal(content, rawAabManifestData)
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, data := range rawAabManifestData.GetElement().GetAttribute() {
+		if data.ResourceId == AndroidVersionCodeId || data.Name == "versionCode" {
+			aabManifestData["versionCode"] = data.GetValue()
+			continue
+		}
+
+		if data.Name == "package" {
+			aabManifestData["applicationId"] = data.GetValue()
+			continue
+		}
+
+		if data.ResourceId == AndroidVersionNameId || data.Name == "versionName" {
+			aabManifestData["versionName"] = data.GetValue()
+			continue
+		}
+	}
+
+	for _, level1 := range rawAabManifestData.GetElement().GetChild() {
+		for _, level2 := range level1.GetElement().GetChild() {
+			if level2.GetElement().GetName() == "meta-data" {
+				if level2.GetElement().GetAttribute()[0].Value == "com.bugsnag.android.API_KEY" {
+					aabManifestData["apiKey"] = level2.GetElement().GetAttribute()[1].Value
+					continue
+				}
+				if level2.GetElement().GetAttribute()[0].Value == "com.bugsnag.android.BUILD_UUID" {
+					aabManifestData["buildUuid"] = level2.GetElement().GetAttribute()[1].Value
+					continue
+				}
+			}
+		}
+	}
+
+	manifestData := &AndroidManifestData{
+		XMLName: xml.Name{
+			Space: "",
+			Local: "manifest",
+		},
+		ApplicationId: aabManifestData["applicationId"],
+		VersionCode:   aabManifestData["versionCode"],
+		VersionName:   aabManifestData["versionName"],
+		Application: AndroidManifestApplicationData{
+			XMLName: xml.Name{
+				Space: "",
+				Local: "application",
+			},
+			MetaData: AndroidManifestMetaData{
+				XMLName: xml.Name{
+					Space: "",
+					Local: "meta-data",
+				},
+				Name:  []string{"com.bugsnag.android.API_KEY", "com.bugsnag.android.BUILD_UUID"},
+				Value: []string{aabManifestData["apiKey"], aabManifestData["buildUuid"]},
+			},
+		},
+	}
+
+	return manifestData, nil
+}
+
+func BuildAndroidInfo(path string) (*AndroidManifestData, error) {
+	//	Workout if its xml or protobuff
```

Comment:
Not sure these comments add anything - the functions they call are self-documenting. Perhaps the func needs an overall comment to summarise intent?